workers/unity/Packages/com.improbable.gdk.transformsynchronization/Systems/PositionSendSystem.cs

```diff
@@ -14,45 +14,63 @@ private struct PositionData
         {
             public readonly int Length;
             public ComponentDataArray<Position.Component> Position;
+            public ComponentDataArray<LastPositionSentData> LastPositionSent;
             [ReadOnly] public ComponentDataArray<Authoritative<Position.Component>> PositionAuthority;
             [ReadOnly] public ComponentDataArray<SpatialEntityId> SpatialEntityIds;
         }
 
         [Inject] private PositionData positionData;
 
-        // Number of position sends per second.
-        private const float SendRateHz = 1.0f;
-
-        private float timeSinceLastSend = 0.0f;
-
         protected override void OnUpdate()
         {
-            // Send update at SendRateHz.
-            timeSinceLastSend += Time.deltaTime;
-            if (timeSinceLastSend < 1.0f / SendRateHz)
+            for (var i = 0; i < positionData.Length; i++)
             {
-                return;
-            }
+                var position = positionData.Position[i];
 
-            timeSinceLastSend = 0.0f;
+                if (position.DirtyBit != true)
+                {
+                    continue;
+                }
 
-            for (var i = 0; i < positionData.Length; i++)
-            {
-                var component = positionData.Position[i];
+                var lastPositionSent = positionData.LastPositionSent[i];
+                lastPositionSent.TimeSinceLastUpdate += Time.deltaTime;
+                positionData.LastPositionSent[i] = lastPositionSent;
 
-                if (component.DirtyBit != true)
+                if (lastPositionSent.TimeSinceLastUpdate <
+                    1.0f / TransformSynchronizationConfig.MaxPositionUpdateRateHz)
                 {
                     continue;
                 }
 
+                var squareDisatnce =
+                    TransformUtils.SquareDistance(lastPositionSent.Position.Coords, positionData.Position[i].Coords);
+
+                if (squareDisatnce == 0.0f)
```

Comment:
Given these are floats - is this condition actually useful?

Reply:
0 is fine. As they will be set from each other this is stable. If they could approach each other in different ways then this wouldn't work well.